identity_stardust/src/document/stardust_document.rs

```diff
@@ -80,10 +61,12 @@ impl StardustDocument {
   // Constructors
   // ===========================================================================
 
-  /// Constructs an empty DID Document with a [`StardustDocument::placeholder_did`] identifier.
+  /// Constructs an empty DID Document with a [`StardustDID::placeholder`] identifier
+  /// for the given `network`.
   // TODO: always take Option<NetworkName> or `new_with_options` for a particular network?
-  pub fn new() -> Self {
-    Self::new_with_id(Self::placeholder_did().clone())
+  // TODO: store the network in the serialized state metadata? Currently it's lost during packing.
```

Comment:
I'm still unsure whether it's good or bad that the network is not embedded. Since the network (other than official names like "main", "dev", etc.) is determined by the client, you could name the network anything and resolve the DID for it. One could change the network name of their private Tangle without affecting the DIDs already on it.

We also have the problem of not matching the network names anymore, e.g. current mainnet has `network_id: "chrysalis-mainnet", bech32_hrp: "iota"`, while current Shimmer alphanet uses `network_name: "alphanet-9", bech32_hrp: "rms"`. Should we switch to the `bech32_hrp` maybe? Then `iota` will be the canonical mainnet that is normalised away, so `did:iota:iota:0x123 = did:iota:0x123`.

Reply:
Good point. In general I like the approach of letting the client determine the network. In theory the HRP sounds nice, but as you pointed out, it's not necessarily unique. It doesn't seem like it would be a problem for the "important" networks we'll be on like `iota` and `smr`; those should be unambiguous. I imagine it's mainly an issue for test networks (i.e. Does `rms` mean the alphanet or the testnet?) where it seems fine to just trust the user to point the client to the right network?

Regardless of hrp or network name, we can set the network name or hrp during resolution so it won't have to be serialized.

Reply:
Okay, will leave the network omitted for now to be determined during resolution. We can revisit the decision if there problems come up. 

The main thing might be inconsistent DIDs within the same network. E.g. Alice signs a credential with their DID `did:iota:alphanet:...` but Bob resolves Alice's DID using a different network like `did:iota:alpha:...`, so either resolution fails (due to unrecognised network) or it is resolved with `alpha` instead of `alphanet` but credential verification fails due to the mismatched network.

Reply:
True that, but the HRP for the network should be the same since it's retrieved from the node, so would this actually occur in practice?

Reply:
Sure, but that's only if we use only an HRP instead of a network name (but again, the HRP has its own problems with potentially matching multiple networks/chains).